internal/resource/config/oauth/oauth_access_token_manager_data_source.go

```
@@ -0,0 +1,380 @@
+			"default_subject_attribute": types.StringPointerValue(attrContract.DefaultSubjectAttribute),
+		}
+		state.AttributeContract, respDiags = types.ObjectValue(attributeContractTypes, attributeContractValues)
+		diags.Append(respDiags...)
+	}
+
+	// state.SelectionSettings
+	if r.SelectionSettings == nil {
+		state.SelectionSettings = types.ObjectNull(selectionSettingsAttrType)
+	} else {
+		state.SelectionSettings, respDiags = types.ObjectValueFrom(ctx, selectionSettingsAttrType, r.SelectionSettings)
+		diags.Append(respDiags...)
+	}
+
+	// state.AccessControlSettings
+	if r.AccessControlSettings == nil {
+		state.AccessControlSettings = types.ObjectNull(accessControlSettingsAttrType)
+	} else {
+		state.AccessControlSettings, respDiags = types.ObjectValueFrom(ctx, accessControlSettingsAttrType, r.AccessControlSettings)
+		diags.Append(respDiags...)
+	}
+
+	// state.SessionValidationSettings
+	if r.SessionValidationSettings == nil {
+		state.SessionValidationSettings = types.ObjectNull(sessionValidationSettingsAttrType)
+	} else {
+		state.SessionValidationSettings, respDiags = types.ObjectValueFrom(ctx, sessionValidationSettingsAttrType, r.SessionValidationSettings)
+		diags.Append(respDiags...)
+	}
+
+	// state.SequenceNumber
+	state.SequenceNumber = types.Int64PointerValue(r.SequenceNumber)
+
+	return diags
+}
+
+// Read resource information
+func (r *oauthAccessTokenManagerDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state oauthAccessTokenManagerDataSourceModel
+
+	diags := req.Config.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiReadOauthAccessTokenManager, httpResp, err := r.apiClient.OauthAccessTokenManagersAPI.GetTokenManager(config.ProviderBasicAuthContext(ctx, r.providerConfig), state.CustomId.ValueString()).Execute()
+
+	if err != nil {
+		if httpResp != nil && httpResp.StatusCode == 404 {
```

I don't think this special handling of 404s applies to data sources. I think a 404 in a data source should probably just return an error.